openreview/venue/venue.py

```diff
@@ -1053,6 +1061,244 @@ def set_SAC_ethics_review_process(self, sac_ethics_flag_duedate=None):
+                    head=submission.id,
+                    tail=iThenticate_submission_id,
+                    label="Created",
+                    weight=-1,
+                )
+
+                iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+                submission_file_binary_data = self.client.get_attachment(
+                    id=submission.id, field_name="pdf"
+                )
+
+                submission_file_object = io.BytesIO(submission_file_binary_data)
+
+                iThenticate_edge.label = "File Sent"
+                iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+                try:
+                    res = iThenticate_client.upload_submission(
+                        submission_id=iThenticate_submission_id,
+                        file_data=submission_file_object,
+                        file_name=submission.content["title"]["value"],
+                    )
+                except Exception as err:
+                    iThenticate_edge.label = "Created"
+                    iThenticate_edge = self.client.post_edge(iThenticate_edge)
+
+            else:
+                print(f"Submission {submission.id} already has edge associated with it with label {edges_dict[submission.id][0]['label']}")
+    
+    def handle_ithenticate_upload_error(self):
+
+        if not self.iThenticate_plagiarism_check:
+            raise openreview.OpenReviewException(
+                "iThenticatePlagiarismCheck is not enabled for this venue."
+            )
+
+        iThenticate_client = openreview.api.iThenticateClient(
+            self.iThenticate_plagiarism_check_api_key,
+            self.iThenticate_plagiarism_check_api_base_url,
+        )
+
+        edges = self.client.get_grouped_edges(
+            invitation=self.get_iThenticate_plagiarism_check_invitation_id(),
+            label="Error_PROCESSING_ERROR",
+            groupby="tail",
+        )
+        
+        for e in tqdm(edges):
+            edge = openreview.api.Edge.from_json(e["values"][0])
+
+            if iThenticate_client.get_submission_status(edge.tail) == "ERROR":
+                # upload error
+                print(f"Uploading submission associated with edge {edge.id} again")
+                submission_file_binary_data = self.client.get_attachment(
+                        id=edge.head, field_name="pdf"
+                )
+
+                submission_file_object = io.BytesIO(submission_file_binary_data)
+
+                edge.label = "File Sent"
+                edge = self.client.post_edge(edge)
+
+                try:
+                    res = iThenticate_client.upload_submission(
+                        submission_id=edge.tail,
+                        file_data=submission_file_object,
+                        file_name=self.client.get_note(id=edge.head).content["title"]["value"],
+                    )
+                except Exception as err:
+                    edge.label = "Created"
```

Comment:
should we go back to original error here?

Reply:
This is for catching 400, or 500 errors. Errors like `Error_CANNOT_EXTRACT_TEXT` won't be caught here. I just realized I don't have something to handle these cases. I can add changes to handle these cases. I can check edges still in the `Created` or `File Uploaded` status and send requests for them again in the handle errors function.

Reply:
In the case of errors like `Error_CANNOT_EXTRACT_TEXT` an exception does not occur

Reply:
You're right

Reply:
I'll make the change